areas/tests/test_address_search_api.py

```diff
@@ -0,0 +1,307 @@
+                "neighbourhood": "Roihuvuori",
+                "neighbourhood_gid": "whosonfirst:neighbourhood:1108727173",
+                "label": "Lumikintie 1, Helsinki",
+            },
+        },
+    ],
+    "bbox": [25.05391, 60.202539, 25.058641, 60.204275],
+}
+
+
+expected_response = {
+    "type": "FeatureCollection",
+    "features": [
+        {
+            "type": "Feature",
+            "geometry": {"type": "Point", "coordinates": [25.057749, 60.202563]},
+            "properties": {
+                "name": "Lumikintie 4",
+            },
+        },
+        {
+            "type": "Feature",
+            "geometry": {"type": "Point", "coordinates": [25.054729, 60.204275]},
+            "properties": {
+                "name": "Lumikintie 5",
+            },
+        },
+        {
+            "type": "Feature",
+            "geometry": {"type": "Point", "coordinates": [25.055386, 60.202539]},
+            "properties": {
+                "name": "Lumikintie 6",
+            },
+        },
+        {
+            "type": "Feature",
+            "geometry": {"type": "Point", "coordinates": [25.05391, 60.20345]},
+            "properties": {
+                "name": "Lumikintie 7",
+            },
+        },
+        {
+            "type": "Feature",
+            "geometry": {"type": "Point", "coordinates": [25.058641, 60.203472]},
+            "properties": {
+                "name": "Lumikintie 1",
+            },
+        },
+    ],
+}
+
+
+@pytest.mark.parametrize(
+    "query_string, expected_digitransit_params",
+    [
+        ("text=lumikintie&language=fi", {"text": "lumikintie", "lang": "fi"}),
+        (
+            "text=snövits väg&language=sv",
+            {"text": "snövits väg", "lang": "sv"},
+        ),
+        ("text=lumikintie", {"text": "lumikintie", "lang": "fi"}),
+    ],
+)
+def test_address_search_returns_correct_response_when_digitransit_request_successful(
+    api_client, mocked_responses, query_string, expected_digitransit_params
+):
+    mocked_responses.get(
+        MOCK_DIGITRANSIT_ADDRESS_SEARCH_URL,
+        match=[
+            matchers.query_param_matcher(
+                COMMON_EXPECTED_DIGITRANSIT_PARAMS | expected_digitransit_params
+            ),
+            matchers.header_matcher(
+                {"digitransit-subscription-key": MOCK_DIGITRANSIT_API_KEY}
+            ),
+        ],
+        json=MOCK_RESPONSE,
+        status=200,
+    )
+
+    response = get(api_client, f"{ADDRESS_SEARCH_URL}?{query_string}")
+    assert response == expected_response
+
+
+def test_address_search_returns_error_when_digitransit_responds_http_error(
+    api_client, mocked_responses
+):
+    mocked_responses.get(
+        MOCK_DIGITRANSIT_ADDRESS_SEARCH_URL,
+        json={"error": "some-unknown-error"},
+        status=503,
+    )
+
+    with pytest.raises(DigitransitApiError):
+        get(api_client, ADDRESS_SEARCH_URL + "?text=lumikintie&language=fi")
```

Comment:
I think you could use `data` parameter here (and with other similar `get` requests) instead of a query string, e.g.
```suggestion
        get(api_client, ADDRESS_SEARCH_URL, data={"text": "lumikintie", "language": "fi"})
```
Not 100% sure if it works, but the underlying code seems to use [Django's test client](https://docs.djangoproject.com/en/4.2/topics/testing/tools/#django.test.Client.get). Either way, up to you if you wanna change this or not (*if* it works)